satellite/gracefulexit/pathcollector.go

```diff
@@ -54,15 +57,24 @@ func (collector *PathCollector) Flush(ctx context.Context) (err error) {
 
 // RemoteSegment takes a remote segment found in metainfo and creates a graceful exit transfer queue item if it doesn't exist already
 func (collector *PathCollector) RemoteSegment(ctx context.Context, path metainfo.ScopedPath, pointer *pb.Pointer) (err error) {
-	if len(collector.nodeIDs) == 0 {
+	if len(collector.nodeIDStorage) == 0 {
 		return nil
 	}
 
+	collector.nodeIDMutex.Lock()
+	defer collector.nodeIDMutex.Unlock()
```

Comment:
nit - maybe we can unlock before the `flush` call?

Reply:
I just noticed that the `flush` is called inside the for loop?

Reply:
Then we would have to make sure to unlock in every case that we could return inside the for loop. I am in favor of keeping the defer even if it means slightly slower processing. But I'm open to changing it if you think we should.